### PR TITLE
Make StorageDescriptor optional for Managed Delta Lake tables

### PR DESCRIPTION
**Pull request by homar**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

a fix
> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

a delta lake connector
> How would you describe this change to a non-technical end user or system administrator?

for managed tables one less parameter is needed

## Related issues, pull requests, and links
Fixes: https://github.com/trinodb/trino/issues/12180

## Documentation

(.) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/HiveMetastoreBackedDeltaLakeMetastore.java [hunk 1]
```diff
@@ -186,12 +187,19 @@ public ProtocolEntry getProtocol(ConnectorSession session, TableSnapshot tableSn
...
                 .getStorage().getSerdeParameters();
         String location = serdeParameters.get(PATH_PROPERTY);
         if (location == null) {
-            throw new TrinoException(DELTA_LAKE_INVALID_SCHEMA, format("No %s property defined for table: %s", PATH_PROPERTY, table));
+            Database database = getDatabase(schemaName).orElseThrow(() -> new SchemaNotFoundException(schemaName));
+            if (database.getLocation().isEmpty()) {
+                throw new TrinoException(DELTA_LAKE_INVALID_SCHEMA, format("No %s property defined for table: %s", PATH_PROPERTY, table));
+            }
+            else {
+                return database.getLocation().get();
```
Comment (findepi) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/HiveMetastoreBackedDeltaLakeMetastore.java:201]: database location != table location

**Comment by findepi** [comment]
I don't get this change. We set location in a table's metadata, so why not use it?
the PR seems to add support for tables with implicit location, ones not created by us. Who does create them?

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/HiveMetastoreBackedDeltaLakeMetastore.java (hunk 1 above)
Comment (homar) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/HiveMetastoreBackedDeltaLakeMetastore.java:201]: I forgot about 1 quite important line...

**Comment by homar** [comment]
> I don't get this change. We set location in a table's metadata, so why not use it? the PR seems to add support for tables with implicit location, ones not created by us. Who does create them?

Probably I misunderstood something. The idea was to derive location from Schema's location for managed tables. Maybe I should add check to verify that this table is in fact managed but I assumed that it makes sense to default to this behaviour. 
@findepi Am I totally wrong?

**Comment by findepi** [comment]
Asked question in the issue: https://github.com/trinodb/trino/issues/12180#issuecomment-1123678473

**Comment by homar** [comment]
Closing as won't do

**Pull request closed without merging**